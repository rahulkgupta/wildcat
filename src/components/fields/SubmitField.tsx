import SubmitField from '@src/util/form/fields/submit';

interface Props {
  field: SubmitField;
  submit: Function;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const SubmitFieldView = (props: Props) => {
  const field = props.field;
  return (
    <input
      type="submit"
      value={field.getValue()}
      onClick={(event) => {
        props.submit();
      }}
      data-testid={field.getId()}
    ></input>
  );
};

export default SubmitFieldView;
