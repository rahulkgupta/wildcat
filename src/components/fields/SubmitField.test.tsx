import SubmitFieldView from './SubmitField';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import SubmitField from '@src/util/form/fields/submit';
import { render, fireEvent } from '@testing-library/react';

describe('SubmitFieldView', () => {
  test('render', () => {
    const submitFieldView: ReactTestRenderer = renderer.create(
      <SubmitFieldView
        {...{
          field: new SubmitField(
            '1234',
            {
              value: 'value',
            },
            jest.fn(),
          ),
          submit: jest.fn(),
        }}
      />,
    );
    expect(submitFieldView.toJSON()).toMatchSnapshot();
  });

  test('onClick', () => {
    const submit: Function = jest.fn();

    const { getByTestId } = render(
      <SubmitFieldView
        {...{
          field: new SubmitField(
            '1234',
            {
              value: 'value',
            },
            jest.fn(),
          ),
          submit,
        }}
      />,
    );
    fireEvent.click(getByTestId('1234'));
    expect(submit).toHaveBeenCalled();
  });
});
