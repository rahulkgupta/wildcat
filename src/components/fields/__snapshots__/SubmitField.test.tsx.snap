// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitFieldView render 1`] = `
<input
  data-testid="1234"
  onClick={[Function]}
  type="submit"
  value="value"
/>
`;
